src/tls_wrap.cc

```diff
@@ -747,6 +750,10 @@ void TLSWrap::SetVerifyMode(const FunctionCallbackInfo<Value>& args) {
 void TLSWrap::EnableSessionCallbacks(
     const FunctionCallbackInfo<Value>& args) {
   TLSWrap* wrap = Unwrap<TLSWrap>(args.Holder());
+  if (wrap->ssl_ == nullptr) {
+    return wrap->env()->ThrowTypeError(
+        "EnableSessionCallbacks after destroySSL");
```

Comment:
I locally created a patch which fixes all the inconsistent error messages like this, in the js files. And I had an internal/errors.js file which defined `ValueError`. Looks like that solves most of the wrong data error messages. But I am not sure if it is worthy introducing a new Error type.